tests/std/tests/P0355R7_calendars_and_time_zones_zoned_time/test.cpp

```diff
@@ -0,0 +1,351 @@
+
+        // (7) zoned_time(string_view name, const sys_time<Duration>& st);
+        assert((zoned_time<seconds>{syd.name, syd.sys}) == zone);
+    }
+    {
+        // (8) zoned_time(TimeZonePtr z, const local_time<Duration>& tp);
+        zoned_time<seconds> zone{&*syd.tz, syd.local};
+        assert(zone.get_time_zone() == syd.tz);
+        assert(zone.get_sys_time() == syd.sys);
+
+        // (9) zoned_time(string_view name, const local_time<Duration>& st);
+        assert((zoned_time<seconds>{syd.name, syd.local}) == zone);
+    }
+    {
+        // (10) zoned_time(TimeZonePtr z, const local_time<Duration>& tp, choose c);
+        zoned_time<seconds> zone{&*syd.tz, syd.local, choose::earliest};
+        assert(zone.get_time_zone() == syd.tz);
+        assert(zone.get_sys_time() == syd.sys);
+
+        // (11) zoned_time(string_view name, const local_time<Duration>& tp, choose c);
+        assert((zoned_time<seconds>{syd.name, syd.local, choose::earliest}) == zone);
+    }
+    {
+        // (12) template<class Duration2, class TimeZonePtr2>
+        //      zoned_time(TimeZonePtr z, const zoned_time<Duration2, TimeZonePtr2>& y);
+        zoned_time<seconds> z1{&*utc.tz, syd.zone};
+        zoned_time<seconds> z2{&*syd.tz, utc.zone};
+        assert(z1.get_time_zone() == utc.tz);
+        assert(z1.get_sys_time() == syd.sys);
+        assert(z2.get_time_zone() == syd.tz);
+        assert(z2.get_sys_time() == utc.sys);
+
+        // (13) template<class Duration2, class TimeZonePtr2>
+        //      zoned_time(TimeZonePtr z, const zoned_time<Duration2, TimeZonePtr2>& y, choose);
+        assert((zoned_time<seconds>{&*utc.tz, syd.zone, choose::earliest}) == z1);
+        assert((zoned_time<seconds>{&*syd.tz, utc.zone, choose::earliest}) == z2);
+
+        // (14) template<class Duration2, class TimeZonePtr2>
+        //      zoned_time(string_view name, const zoned_time<Duration2, TimeZonePtr2>& y);
+        assert((zoned_time<seconds>{utc.name, syd.zone}) == z1);
+        assert((zoned_time<seconds>{syd.name, utc.zone}) == z2);
+
+        // (15) template<class Duration2, class TimeZonePtr2>
+        //      zoned_time(string_view name, const zoned_time<Duration2, TimeZonePtr2>& y, choose);
+        assert((zoned_time<seconds>{utc.name, syd.zone, choose::earliest}) == z1);
+        assert((zoned_time<seconds>{syd.name, utc.zone, choose::earliest}) == z2);
+
+        // when (_Duration2 != _Duration)
+        zoned_time<hours> zone_hours{&*syd.tz, sys_time<hours>{hours{1}}};
+        zoned_time<seconds> zone{&*utc.tz, zone_hours};
+        assert(zone.get_time_zone() == utc.tz);
+        assert(zone.get_sys_time() == sys_time<seconds>{hours{1}});
+
+        assert((zoned_time<seconds>{&*utc.tz, zone_hours, choose::earliest}) == zone);
+        assert((zoned_time<seconds>{utc.name, zone_hours}) == zone);
+        assert((zoned_time<seconds>{utc.name, zone_hours, choose::earliest}) == zone);
+    }
+}
+
+void zonetime_operator_test() {

```

Comment:
Are we testing `operator sys_time<duration>()` and `explicit operator local_time<duration>()` anywhere?

Reply:
No, I will add those here.